__test__/users.test.js

```diff
@@ -145,33 +145,43 @@ describe('GET /users', () => {
         expect(res.body.message).toBe('Token is required');
       });
   });
-
-  // it('should return 401 if token is invalid', () => {
-  //   return request(server)
-  //     .patch('/api/user/:damola/profile')
-  //     .set({ token: 'gdgfhhrbgegq2ehnfnsnjthrtn' })
-  //     .send()
-  //     .then(res => {
-  //       expect(res.status).toBe(401);
-  //       expect(res.body.message).toBe('Error token type');
-  //     });
-  // });
 });
 
-// describe('PATCH /:username/profile', () => {
-//   it('should return an object if the body is created', () => {
-//     return request(server)
-//       .patch('api/user/:damola/profile')
-//       .send({
-//         firstName: 'Damola',
-//         lastName: 'Adewunmi',
-//         eMail: 'damolasd@gmail.com',
-//         bio: 'Lambda School',
-//         locatiionId: 20
-//       })
-//       .then(res => {
-//         expect(res.status).toBe(201);
-//         expect(res.body).toBe(Object);
-//       });
-//   });
-// });
+describe('GET user profile information', () => {
+  it('should return a 200 code', async () => {
+    const user = {
+      id: 1,
+      username: 'john'
+    };
+    const jwtToken = await jwt.generateToken(user);
+    return request(server)
+      .get('/api/user/john/profile')
+      .set({ token: jwtToken })
+      .then(res => {
+        expect(res.status).toBe(200);
+      });
+  });
+
+  it('should return a 401 code if token does not match user', async () => {
+    const user = {
+      id: 1,
+      username: 'john'
+    };
+    const jwtToken = await jwt.generateToken(user);
+    return request(server)
+      .get('/api/user/damola/profile')
+      .set({ token: jwtToken })
+      .then(res => {
+        expect(res.status).toBe(401);
+      });
+  });
+
+  it('should return 401 if no token is provided', () => {
+    return request(server)
+      .get('/api/user/john/profile')
+      .then(res => {
+        expect(res.status).toBe(401);
+        expect(res.body.message).toBe('Token is required');
+      });
```

Comment:
can you also write the test that validates that the user providing the token is the user accessing it????
check-in the helper folder the jwt file and auth user function

Reply:
Wrote the tests